<!DOCTYPE html>



  


<html class="theme-next gemini use-motion" lang="zh-Hans">
<head>
  <meta charset="UTF-8"/>
<meta http-equiv="X-UA-Compatible" content="IE=edge" />
<meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1"/>
<meta name="theme-color" content="#222">









<meta http-equiv="Cache-Control" content="no-transform" />
<meta http-equiv="Cache-Control" content="no-siteapp" />
















  
  
  <link href="/lib/fancybox/source/jquery.fancybox.css?v=2.1.5" rel="stylesheet" type="text/css" />




  
  
  
  

  
    
    
  

  

  

  

  

  
    
    
    <link href="//fonts.googleapis.com/css?family=Lato:300,300italic,400,400italic,700,700italic&subset=latin,latin-ext" rel="stylesheet" type="text/css">
  






<link href="/lib/font-awesome/css/font-awesome.min.css?v=4.6.2" rel="stylesheet" type="text/css" />

<link href="/css/main.css?v=5.1.2" rel="stylesheet" type="text/css" />


  <meta name="keywords" content="Hexo, NexT" />








  <link rel="shortcut icon" type="image/x-icon" href="/favicon.ico?v=5.1.2" />






<meta property="og:type" content="website">
<meta property="og:title" content="The Future Is In Our Hands">
<meta property="og:url" content="http://gjq0724.github.io/index.html">
<meta property="og:site_name" content="The Future Is In Our Hands">
<meta property="og:locale" content="zh-Hans">
<meta name="twitter:card" content="summary">
<meta name="twitter:title" content="The Future Is In Our Hands">



<script type="text/javascript" id="hexo.configurations">
  var NexT = window.NexT || {};
  var CONFIG = {
    root: '/',
    scheme: 'Gemini',
    version: '5.1.2',
    sidebar: {"position":"left","display":"post","offset":12,"offset_float":12,"b2t":false,"scrollpercent":false,"onmobile":false},
    fancybox: true,
    tabs: true,
    motion: true,
    duoshuo: {
      userId: '0',
      author: '博主'
    },
    algolia: {
      applicationID: '',
      apiKey: '',
      indexName: '',
      hits: {"per_page":10},
      labels: {"input_placeholder":"Search for Posts","hits_empty":"We didn't find any results for the search: ${query}","hits_stats":"${hits} results found in ${time} ms"}
    }
  };
</script>



  <link rel="canonical" href="http://gjq0724.github.io/"/>





  <title>The Future Is In Our Hands</title>
  








</head>

<body itemscope itemtype="http://schema.org/WebPage" lang="zh-Hans">

  
  
    
  

  <div class="container sidebar-position-left 
  page-home">
    <div class="headband"></div>
    <a href="https://github.com/gjq0724/gjq0724.github.io"><img style="position: absolute; top: 0; right: 0; border: 0;" src="https://camo.githubusercontent.com/38ef81f8aca64bb9a64448d0d70f1308ef5341ab/68747470733a2f2f73332e616d617a6f6e6177732e636f6d2f6769746875622f726962626f6e732f666f726b6d655f72696768745f6461726b626c75655f3132313632312e706e67" alt="Fork me on GitHub" data-canonical-src="https://s3.amazonaws.com/github/ribbons/forkme_right_darkblue_121621.png"></a>
    <header id="header" class="header" itemscope itemtype="http://schema.org/WPHeader">
      <div class="header-inner"><div class="site-brand-wrapper">
  <div class="site-meta ">
    

    <div class="custom-logo-site-title">
      <a href="/"  class="brand" rel="start">
        <span class="logo-line-before"><i></i></span>
        <span class="site-title">The Future Is In Our Hands</span>
        <span class="logo-line-after"><i></i></span>
      </a>
    </div>
      
        <p class="site-subtitle"></p>
      
  </div>

  <div class="site-nav-toggle">
    <button>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
      <span class="btn-bar"></span>
    </button>
  </div>
</div>

<nav class="site-nav">
  

  
    <ul id="menu" class="menu">
      
        
        <li class="menu-item menu-item-home">
          <a href="/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-home"></i> <br />
            
            首页
          </a>
        </li>
      
        
        <li class="menu-item menu-item-categories">
          <a href="/categories/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-th"></i> <br />
            
            分类
          </a>
        </li>
      
        
        <li class="menu-item menu-item-about">
          <a href="/about/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-user"></i> <br />
            
            关于
          </a>
        </li>
      
        
        <li class="menu-item menu-item-tags">
          <a href="/tags/" rel="section">
            
              <i class="menu-item-icon fa fa-fw fa-tags"></i> <br />
            
            标签
          </a>
        </li>
      

      
    </ul>
  

  
</nav>



 </div>
    </header>

    <main id="main" class="main">
      <div class="main-inner">
        <div class="content-wrap">
          <div id="content" class="content">
            
  <section id="posts" class="posts-expand">
    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://gjq0724.github.io/2017/08/24/methDiffRegion/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Guo Junqiang">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="The Future Is In Our Hands">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2017/08/24/methDiffRegion/" itemprop="url">methDiffRegion</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2017-08-24T16:39:55+08:00">
                2017-08-24
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p>##bismark处理测序数据:</p>
<ol>
<li><p>对拟南芥基因组建索引</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line">bismark_genome_preparation   --verbose     --bowtie2  /mnt/S30/guojq/methylation/genome   </div><div class="line">/*将tair10.fa放到/mnt/S30/guojq/methylation/genome文件夹，bismark会在这个目录下建立索引，之后使用该路径调用索引文件*/</div></pre></td></tr></table></figure>
</li>
<li><p>bismark处理测序数据</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div></pre></td><td class="code"><pre><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/CIBC-CK1/CIBC-CK1_1.fq -2 /mnt/S30/guojq/methylation/data/CIBC-CK1/CIBC-CK1_2.fq  -o ./CIBC-CK1 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/CIBC-CK2/CIBC-CK2_1.fq -2 /mnt/S30/guojq/methylation/data/CIBC-CK2/CIBC-CK2_2.fq  -o ./CIBC-CK2 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/CIBC-DT1/CIBC-DT1_1.fq -2 /mnt/S30/guojq/methylation/data/CIBC-DT1/CIBC-DT1_2.fq  -o ./CIBC-DT1 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/CIBC-DT2/CIBC-DT2_1.fq -2 /mnt/S30/guojq/methylation/data/CIBC-DT2/CIBC-DT2_2.fq  -o ./CIBC-DT2 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/BURO-CK1/BURO-CK1_1.fq -2 /mnt/S30/guojq/methylation/data/BURO-CK1/BURO-CK1_2.fq  -o ./BURO-CK1 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/BURO-CK2/BURO-CK2_1.fq -2 /mnt/S30/guojq/methylation/data/BURO-CK2/BURO-CK2_2.fq  -o ./BURO-CK2 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/BURO-DT1/BURO-DT1_1.fq -2 /mnt/S30/guojq/methylation/data/BURO-DT1/BURO-DT1_2.fq  -o ./BURO-DT1 &amp;</div><div class="line">bismark -p 5  /mnt/S30/guojq/methylation/genome --bowtie2  -1 /mnt/S30/guojq/methylation/data/BURO-DT2/BURO-DT2_1.fq -2 /mnt/S30/guojq/methylation/data/BURO-DT2/BURO-DT2_2.fq  -o ./BURO-DT2</div></pre></td></tr></table></figure>
</li>
</ol>
<hr>
<p>##methylKit分析甲基化差异性表达</p>
<ol>
<li><p>对<code>bismark</code>生成的bam文件排序，转化为sam文件，命令:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div></pre></td><td class="code"><pre><div class="line">foreach BAM(*_pe.bam)</div><div class="line">samtools sort <span class="variable">$BAM</span> `\basename <span class="variable">$BAM</span> .bam`.sorted</div><div class="line">samtools view -h `\basename <span class="variable">$BAM</span> .bam`.sorted.bam &gt; `\basename <span class="variable">$BAM</span> .bam`.sorted.sam</div><div class="line">end</div></pre></td></tr></table></figure>
</li>
<li><p>根据sam文件提取甲基化信息，命令如下:</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">library</span>(methylKit)</div><div class="line"></div><div class="line">fns=Sys.glob(<span class="string">'*.sam'</span>)</div><div class="line"><span class="keyword">for</span>(fn <span class="keyword">in</span> fns)&#123;</div><div class="line">			wri_f=paste(   strsplit( fn, <span class="string">'_1_bismark_bt2_pe.sorted.sam'</span>, fixed=<span class="literal">T</span> )[[<span class="number">1</span>]][<span class="number">1</span>] ,<span class="string">''</span>, sep=<span class="string">''</span>)</div><div class="line">			myobj=read.bismark(fn,wri_f,assembly=<span class="string">"tair10"</span>,save.context=<span class="string">"CpG"</span>,read.context=<span class="string">"CpG"</span>,save.fold=getwd())</div><div class="line">		&#125;</div></pre></td></tr></table></figure>
</li>
</ol>
<p><code>read.bismark</code>是<code>methylKit</code>自带的一个方法，可以提取CpG、CHG、CHH甲基化水平，生成包含甲基化水平的txt文件，格式如下：<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">chrBase|chr|base|strand|coverage|freqC|freqT</div><div class="line">Chr1.500|Chr1|500|F|20|0.00|100.00</div><div class="line">Chr1.511|Chr1|511|F|22|0.00|100.00</div></pre></td></tr></table></figure></p>
<ol>
<li><p>根据甲基化水平文件，生成<code>bedGraph</code>文件</p>
<figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">library</span>(methylKit)</div><div class="line">fns=Sys.glob(<span class="string">'*.txt'</span>)</div><div class="line">		<span class="keyword">for</span>(fn <span class="keyword">in</span> fns)&#123;</div><div class="line">			wri_f=paste(   strsplit( fn, <span class="string">'_CpG.txt'</span>, fixed=<span class="literal">T</span> )[[<span class="number">1</span>]][<span class="number">1</span>] , <span class="string">''</span>, sep=<span class="string">''</span>)</div><div class="line">			myobj=methRead(fn,wri_f,<span class="string">"tair10"</span>)</div><div class="line">			wri_f=paste(   strsplit( fn, <span class="string">'.txt'</span>, fixed=<span class="literal">T</span> )[[<span class="number">1</span>]][<span class="number">1</span>] , <span class="string">'.bedgraph'</span>, sep=<span class="string">''</span>)</div><div class="line">			bedgraph(myobj,wri_f,col.name=<span class="string">"perc.meth"</span>,unmeth=<span class="literal">FALSE</span>,log.transform=<span class="literal">FALSE</span>,negative=<span class="literal">FALSE</span>,add.on=<span class="string">""</span>)</div><div class="line"></div><div class="line">		&#125;</div></pre></td></tr></table></figure>
</li>
<li><p>根据甲基化水平文件，计算甲基化差异，找差异表达区域</p>
</li>
</ol>
<figure class="highlight r"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div></pre></td><td class="code"><pre><div class="line"><span class="keyword">library</span>(methylKit)</div><div class="line">file.list1 &lt;- list(<span class="string">"BURO-CK1_CpG.txt"</span>,<span class="string">"BURO-CK2_CpG.txt"</span>,<span class="string">"BURO-DT1_CpG.txt"</span>,<span class="string">"BURO-DT2_CpG.txt"</span>) </div><div class="line">file.list2 &lt;- list(<span class="string">"CIBC-CK1_CpG.txt"</span>,<span class="string">"CIBC-CK2_CpG.txt"</span>,<span class="string">"CIBC-DT1_CpG.txt"</span>,<span class="string">"CIBC-DT2_CpG.txt"</span>)</div><div class="line">objs1 &lt;- methRead(file.list1,sample.id=list(<span class="string">"BURO-CK1"</span>,<span class="string">"BURO-CK2"</span>,<span class="string">"BURO-DT1"</span>,<span class="string">"BURO-DT2"</span>),assembly=<span class="string">"tair10"</span>,treatment=c(<span class="number">0</span>,<span class="number">0</span>,<span class="number">1</span>,<span class="number">1</span>))</div><div class="line">objs2 &lt;- methRead(file.list2,sample.id=list(<span class="string">"CIBC-CK1"</span>,<span class="string">"CIBC-CK2"</span>,<span class="string">"CIBC-DT1"</span>,<span class="string">"CIBC-DT2"</span>),assembly=<span class="string">"tair10"</span>,treatment=c(<span class="number">0</span>,<span class="number">0</span>,<span class="number">1</span>,<span class="number">1</span>))</div><div class="line">regionCounts1 &lt;- tileMethylCounts(objs1 , win.size = <span class="number">500</span>,step.size = <span class="number">500</span>)</div><div class="line">regionCounts2 &lt;- tileMethylCounts(objs2 , win.size = <span class="number">500</span>,step.size = <span class="number">500</span>)</div><div class="line">meth1 &lt;- unite(regionCounts1)</div><div class="line">meth2 &lt;- unite(regionCounts2)</div><div class="line">write.table(meth1,<span class="string">"BURO-regionCounts.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span>)</div><div class="line">write.table(meth2,<span class="string">"CIBC-regionCounts.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span>)</div><div class="line">diff1 &lt;- calculateDiffMeth(meth1)</div><div class="line">diff2 &lt;- calculateDiffMeth(meth2)</div><div class="line">myDiff1.hyper &lt;- getMethylDiff(diff1,difference=<span class="number">25</span>,qvalue=<span class="number">0.01</span>,type=<span class="string">"hyper"</span>)</div><div class="line">myDiff1.hypo &lt;- getMethylDiff(diff1,difference=<span class="number">25</span>,qvalue=<span class="number">0.01</span>,type=<span class="string">"hypo"</span>)</div><div class="line">write.table(myDiff1.hyper,<span class="string">"BURO-hyper-diff-region.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span>)</div><div class="line">write.table(myDiff1.hypo,<span class="string">"BURO-hypo-diff-region.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span>)</div><div class="line">myDiff2.hyper &lt;- getMethylDiff(diff2,difference=<span class="number">25</span>,qvalue=<span class="number">0.01</span>,type=<span class="string">"hyper"</span>)</div><div class="line">myDiff2.hypo &lt;- getMethylDiff(diff2,difference=<span class="number">25</span>,qvalue=<span class="number">0.01</span>,type=<span class="string">"hypo"</span>)</div><div class="line">)</div><div class="line">write.table(myDiff2.hyper,<span class="string">"CIBC-hyper-diff-region.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span>)</div><div class="line">write.table(myDiff2.hypo,<span class="string">"CIBC-hypo-diff-region.txt"</span>,sep=<span class="string">"\t"</span>,row.names=<span class="literal">FALSE</span>,col.names=<span class="literal">TRUE</span></div></pre></td></tr></table></figure>
          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
      

  

  
  
  

  <article class="post post-type-normal" itemscope itemtype="http://schema.org/Article">
  
  
  
  <div class="post-block">
    <link itemprop="mainEntityOfPage" href="http://gjq0724.github.io/2017/08/24/gff2bed/">

    <span hidden itemprop="author" itemscope itemtype="http://schema.org/Person">
      <meta itemprop="name" content="Guo Junqiang">
      <meta itemprop="description" content="">
      <meta itemprop="image" content="/images/avatar.gif">
    </span>

    <span hidden itemprop="publisher" itemscope itemtype="http://schema.org/Organization">
      <meta itemprop="name" content="The Future Is In Our Hands">
    </span>

    
      <header class="post-header">

        
        
          <h1 class="post-title" itemprop="name headline">
                
                <a class="post-title-link" href="/2017/08/24/gff2bed/" itemprop="url">gff2bed</a></h1>
        

        <div class="post-meta">
          <span class="post-time">
            
              <span class="post-meta-item-icon">
                <i class="fa fa-calendar-o"></i>
              </span>
              
                <span class="post-meta-item-text">发表于</span>
              
              <time title="创建于" itemprop="dateCreated datePublished" datetime="2017-08-24T16:12:54+08:00">
                2017-08-24
              </time>
            

            

            
          </span>

          

          
            
          

          
          

          

          

          

        </div>
      </header>
    

    
    
    
    <div class="post-body" itemprop="articleBody">

      
      

      
        
          
            <p><em>BED 文件格式提供了一种灵活的方式来定义的数据行，以用来描述注释的信息。BED行有3个必须的列和9个额外可选的列。 每行的数据格式要求一致。通常所见的bed文件为6列，但是一些软件要求输入12列bed文件(UCSC  Genome Browser)，通过探索研究出一下两种从gff3/gtf转为12列bed文件的方法。</em></p>
<p>##GFFtools-GX</p>
<p><code>GFFtools-GX</code>是<code>vipints</code>分享在<a href="https://github.com/vipints/GFFtools-GX.git" target="_blank" rel="external">GitHub</a>上的一个程序集，里面包含了gff、gtf、gbk、bed格式之间的相互转换。</p>
<ul>
<li><p>在<code>GitHub</code>上，<code>vipints</code>分享了一种使用方式，该方式需要在本地先配置<code>Galaxy Tool shed</code>(具体配置看<a href="https://toolshed.g2.bx.psu.edu/" target="_blank" rel="external">网站说明</a>，并安装了python2.6或者2.7以及biopython。配置完成后，使用一下命令调用该软件包:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">sh run_functional_tests.sh -id fml_gtf2gff</div></pre></td></tr></table></figure>
</li>
<li><p>另一种方式是将软件包下载下来，直接调用软件包中的脚本:</p>
<figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line"><span class="comment">#Convert genome annotation data in GFF/GTF to a 12 column BED format. BED format typically represents the transcript models. </span></div><div class="line">Usage: python gff_to_bed.py in.gff &gt; out.bed</div><div class="line">       python gtf_to_bed.py in.gtf &gt; out.bed</div></pre></td></tr></table></figure>
</li>
</ul>
<p>直接下载的软件包中的<code>gff_to_bed.py</code>在运行时会出现以下报错，<br><figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">TypeError: len() of unsized object</div></pre></td></tr></table></figure></p>
<p>在<code>GitHub</code>上<code>tzintzuni</code>给出了解决方法，修改<code>gff_to_bed.py</code>文件，将以下代码<br><figure class="highlight python"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div></pre></td><td class="code"><pre><div class="line"><span class="function"><span class="keyword">def</span> <span class="title">writeBED</span><span class="params">(tinfo)</span>:</span></div><div class="line">    <span class="string">"""</span></div><div class="line"><span class="string">    writing result files in bed format </span></div><div class="line"><span class="string">    @args tinfo: list of genes </span></div><div class="line"><span class="string">    @type tinfo: numpy object  </span></div><div class="line"><span class="string">    """</span></div><div class="line"></div><div class="line">    <span class="keyword">for</span> ent1 <span class="keyword">in</span> tinfo:</div><div class="line">        child_flag = <span class="keyword">False</span>  </div><div class="line"></div><div class="line">        <span class="keyword">for</span> idx, tid <span class="keyword">in</span> enumerate(ent1[<span class="string">'transcripts'</span>]):</div><div class="line">            child_flag = <span class="keyword">True</span> </div><div class="line">            exon_cnt = len(ent1[<span class="string">'exons'</span>][idx])</div><div class="line">            exon_len = <span class="string">''</span></div><div class="line">            exon_cod = <span class="string">''</span> </div><div class="line">            rel_start = <span class="keyword">None</span> </div><div class="line">            rel_stop = <span class="keyword">None</span> </div><div class="line">            <span class="keyword">for</span> idz, ex_cod <span class="keyword">in</span> enumerate(ent1[<span class="string">'exons'</span>][idx]):<span class="comment">#check for exons of corresponding transcript  </span></div><div class="line">                exon_len += <span class="string">'%d,'</span> % (ex_cod[<span class="number">1</span>]-ex_cod[<span class="number">0</span>]+<span class="number">1</span>)</div><div class="line">                <span class="keyword">if</span> idz == <span class="number">0</span>: <span class="comment">#calculate the relative start position </span></div><div class="line">                    exon_cod += <span class="string">'0,'</span></div><div class="line">                    rel_start = int(ex_cod[<span class="number">0</span>])<span class="number">-1</span> </div><div class="line">                    rel_stop = int(ex_cod[<span class="number">1</span>])</div><div class="line">                <span class="keyword">else</span>:</div><div class="line">                    exon_cod += <span class="string">'%d,'</span> % (ex_cod[<span class="number">0</span>]<span class="number">-1</span>-rel_start) <span class="comment">## shifting the coordinates to zero </span></div><div class="line">                    rel_stop = int(ex_cod[<span class="number">1</span>])</div></pre></td></tr></table></figure></p>
<p>替换为<br><figure class="highlight python"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div><div class="line">6</div><div class="line">7</div><div class="line">8</div><div class="line">9</div><div class="line">10</div><div class="line">11</div><div class="line">12</div><div class="line">13</div><div class="line">14</div><div class="line">15</div><div class="line">16</div><div class="line">17</div><div class="line">18</div><div class="line">19</div><div class="line">20</div><div class="line">21</div><div class="line">22</div><div class="line">23</div><div class="line">24</div><div class="line">25</div><div class="line">26</div><div class="line">27</div><div class="line">28</div><div class="line">29</div></pre></td><td class="code"><pre><div class="line"><span class="function"><span class="keyword">def</span> <span class="title">writeBED</span><span class="params">(tinfo)</span>:</span></div><div class="line">    <span class="string">"""</span></div><div class="line"><span class="string">    writing result files in bed format </span></div><div class="line"><span class="string"></span></div><div class="line"><span class="string">    @args tinfo: list of genes </span></div><div class="line"><span class="string">    @type tinfo: numpy object  </span></div><div class="line"><span class="string">    """</span></div><div class="line"></div><div class="line">    <span class="keyword">for</span> ent1 <span class="keyword">in</span> tinfo:</div><div class="line">        child_flag = <span class="keyword">False</span>  </div><div class="line"></div><div class="line">        <span class="keyword">for</span> idx, tid <span class="keyword">in</span> enumerate(ent1[<span class="string">'transcripts'</span>]):</div><div class="line">            child_flag = <span class="keyword">True</span> </div><div class="line">            exon_cnt = <span class="number">0</span></div><div class="line">            exon_len = <span class="string">''</span></div><div class="line">            exon_cod = <span class="string">''</span> </div><div class="line">            rel_start = <span class="keyword">None</span> </div><div class="line">            rel_stop = <span class="keyword">None</span> </div><div class="line">            <span class="keyword">if</span> ent1[<span class="string">'exons'</span>][idx].ndim &gt; <span class="number">0</span>:</div><div class="line">                exon_cnt = len(ent1[<span class="string">'exons'</span>][idx])</div><div class="line">                <span class="keyword">for</span> idz, ex_cod <span class="keyword">in</span> enumerate(ent1[<span class="string">'exons'</span>][idx]):<span class="comment">#check for exons of corresponding transcript  </span></div><div class="line">                    exon_len += <span class="string">'%d,'</span> % (ex_cod[<span class="number">1</span>]-ex_cod[<span class="number">0</span>]+<span class="number">1</span>)</div><div class="line">                    <span class="keyword">if</span> idz == <span class="number">0</span>: <span class="comment">#calculate the relative start position </span></div><div class="line">                        exon_cod += <span class="string">'0,'</span></div><div class="line">                        rel_start = int(ex_cod[<span class="number">0</span>])<span class="number">-1</span> </div><div class="line">                        rel_stop = int(ex_cod[<span class="number">1</span>])</div><div class="line">                    <span class="keyword">else</span>:</div><div class="line">                        exon_cod += <span class="string">'%d,'</span> % (ex_cod[<span class="number">0</span>]<span class="number">-1</span>-rel_start) <span class="comment">## shifting the coordinates to zero </span></div><div class="line">                        rel_stop = int(ex_cod[<span class="number">1</span>])</div></pre></td></tr></table></figure></p>
<p>修改后运行脚本，出现以下错误:<br><figure class="highlight plain"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div></pre></td><td class="code"><pre><div class="line"></div><div class="line"></div></pre></td></tr></table></figure></p>
<p>将脚本末尾以下部分代码:<br><figure class="highlight python"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">out_print = [ent1[<span class="string">'chr'</span>],</div><div class="line">        <span class="string">'%d'</span> % int(ent1[<span class="string">'start'</span>])<span class="number">-1</span>, </div><div class="line">        <span class="string">'%d'</span> % int(ent1[<span class="string">'stop'</span>]),</div></pre></td></tr></table></figure></p>
<p>修改为:<br><figure class="highlight python"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div></pre></td><td class="code"><pre><div class="line">out_print = [ent1[<span class="string">'chr'</span>],</div><div class="line">        <span class="string">'%d'</span> % (int(ent1[<span class="string">'start'</span>])<span class="number">-1</span>), </div><div class="line">        <span class="string">'%d'</span> % int(ent1[<span class="string">'stop'</span>]),</div></pre></td></tr></table></figure></p>
<p>即可解决问题。</p>
<hr>
<p>##BEDOPS<br>BEDOPS包含了多种工具，最常用的就是<code>convert2bed</code>，可将方便的将gff、gtf、bam等多种格式转换为bed或者starch格式，但在转换为12列bed文件时需要借助<code>UCSC</code>提供的软件包来完成。<br>下载需要的UCSC包:<br><figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div><div class="line">2</div><div class="line">3</div><div class="line">4</div><div class="line">5</div></pre></td><td class="code"><pre><div class="line"><span class="comment">#可浏览网址http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/ 查看各种包</span></div><div class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/gff3ToGenePred</div><div class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/gtfToGenePred</div><div class="line">wget -c http://hgdownload.cse.ucsc.edu/admin/exe/linux.x86_64/genePredToBed</div><div class="line">chmod 755 gff3ToGenePred gtfToGenePred genePredToBed</div></pre></td></tr></table></figure></p>
<p>转换gtf文件为bed12:<br><figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">gtfToGenePred test.gtf test.genePhred &amp;&amp; genePredToBed test.genePhred result.bed &amp;&amp; rm test.genePhred</div></pre></td></tr></table></figure></p>
<p>转换gff文件为bed12:<br><figure class="highlight sh"><table><tr><td class="gutter"><pre><div class="line">1</div></pre></td><td class="code"><pre><div class="line">gtf3ToGenePred test.gff test.genePhred &amp;&amp; genePredToBed test.genePhred result.bed &amp;&amp; rm test.genePhred</div></pre></td></tr></table></figure></p>
<hr>
<p>如有问题，欢迎交流。</p>

          
        
      
    </div>
    
    
    

    

    

    

    <footer class="post-footer">
      

      

      

      
      
        <div class="post-eof"></div>
      
    </footer>
  </div>
  
  
  
  </article>


    
  </section>

  


          </div>
          


          

        </div>
        
          
  
  <div class="sidebar-toggle">
    <div class="sidebar-toggle-line-wrap">
      <span class="sidebar-toggle-line sidebar-toggle-line-first"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-middle"></span>
      <span class="sidebar-toggle-line sidebar-toggle-line-last"></span>
    </div>
  </div>

  <aside id="sidebar" class="sidebar">
    
    <div class="sidebar-inner">

      

      

      <section class="site-overview sidebar-panel sidebar-panel-active">
        <div class="site-author motion-element" itemprop="author" itemscope itemtype="http://schema.org/Person">
          <img class="site-author-image" itemprop="image"
               src="/images/avatar.gif"
               alt="Guo Junqiang" />
          <p class="site-author-name" itemprop="name">Guo Junqiang</p>
           
              <p class="site-description motion-element" itemprop="description"></p>
          
        </div>
        <nav class="site-state motion-element">

          
            <div class="site-state-item site-state-posts">
            
              <a href="/archives">
            
                <span class="site-state-item-count">2</span>
                <span class="site-state-item-name">日志</span>
              </a>
            </div>
          

          

          
            
            
            <div class="site-state-item site-state-tags">
              <a href="/tags/index.html">
                <span class="site-state-item-count">2</span>
                <span class="site-state-item-name">标签</span>
              </a>
            </div>
          

        </nav>

        

        <div class="links-of-author motion-element">
          
        </div>

        
        

        
        

        


      </section>

      

      

    </div>
  </aside>


        
      </div>
    </main>

    <footer id="footer" class="footer">
      <div class="footer-inner">
        <script async src="https://dn-lbstatics.qbox.me/busuanzi/2.3/busuanzi.pure.mini.js">
</script>
<div class="copyright" >
  
  &copy; 
  <span itemprop="copyrightYear">2017</span>
  <span class="with-love">
    <i class="fa fa-heart"></i>
  </span>
  <span class="author" itemprop="copyrightHolder">Guo Junqiang</span>

  
</div>



        







        
      </div>
    </footer>

    
      <div class="back-to-top">
        <i class="fa fa-arrow-up"></i>
        
      </div>
    

  </div>

  

<script type="text/javascript">
  if (Object.prototype.toString.call(window.Promise) !== '[object Function]') {
    window.Promise = null;
  }
</script>









  


  











  
  <script type="text/javascript" src="/lib/jquery/index.js?v=2.1.3"></script>

  
  <script type="text/javascript" src="/lib/fastclick/lib/fastclick.min.js?v=1.0.6"></script>

  
  <script type="text/javascript" src="/lib/jquery_lazyload/jquery.lazyload.js?v=1.9.7"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/velocity/velocity.ui.min.js?v=1.2.1"></script>

  
  <script type="text/javascript" src="/lib/fancybox/source/jquery.fancybox.pack.js?v=2.1.5"></script>

  
  <script type="text/javascript" src="/lib/canvas-nest/canvas-nest.min.js"></script>


  


  <script type="text/javascript" src="/js/src/utils.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/motion.js?v=5.1.2"></script>



  
  


  <script type="text/javascript" src="/js/src/affix.js?v=5.1.2"></script>

  <script type="text/javascript" src="/js/src/schemes/pisces.js?v=5.1.2"></script>



  
    <script type="text/javascript" src="/js/src/scrollspy.js?v=5.1.2"></script>
<script type="text/javascript" src="/js/src/post-details.js?v=5.1.2"></script>

  
  
  


  <script type="text/javascript" src="/js/src/bootstrap.js?v=5.1.2"></script>


  
  <script type="text/javascript" src="//cdn.bootcss.com/canvas-nest.js/1.0.0/canvas-nest.min.js"></script>
  
  


  




	





  





  








  





  

  

  

  

  

  

</body>
</html>
